File: python/timer.py
import time


class Timer:

    def __init__(self,  new_id, interval):
        self.id = new_id
        self.interval = interval
        self.elapsed = 0
        self.tick = 0
        self.enabled = False
        self.start = int(round(time.time() * 1000))

    def enable(self):
        if not self.enabled:
            self.enabled = True

    def disable(self):
        if self.enabled:
            self.enabled = False

    # When started, add 1 tick each X interval
    # To use to an update function
    def run(self):
        if self.enabled:
            self.elapsed = int(round(time.time() * 1000)) - self.start
            if self.elapsed > self.interval:
                self.tick += 1
                self.start = int(round(time.time() * 1000))

    # To use to an update function
    def next_tick(self):
        return self.elapsed > self.interval

    def get_tick(self):
        return self.tick

    def reset(self):
        self.elapsed = 0
        self.tick = 0
        self.enabled = False

    def restart(self):
        self.elapsed = 0
        self.tick = 0
        self.enabled = True

File: js/timer.js
class Timer {

    constructor(id, interval) {
        this.id = id;
        this.interval = interval;
        this.elapsed = 0;
        this.tick = 0;
        this.enabled = false;
        this.start = new Date().getTime();
	}
	
    enable() {
        if (!this.enabled) {
            this.enabled = true;
        }
	}
	
    disable() {
        if (this.enabled) {
            this.enabled = false;
        }
	}
	
    // When started, add 1 tick each X interval
    // To use to an update function
    run() {
        if (this.enabled) {
            this.elapsed = new Date().getTime() - this.start;
            if (this.elapsed > this.interval) {
                this.tick++;
                this.start = new Date().getTime();
            }
        }
	}
	
    // To use to an update function
    nextTick() {
        return this.elapsed > this.interval;
	}
	
    reset() {
        this.elapsed = 0;
        this.tick = 0;
        this.enabled = false;
	}
	
    restart() {
        this.elapsed = 0;
        this.tick = 0;
        this.enabled = true;
    }
}